src/pytorch_lightning/strategies/parallel.py

```diff
@@ -89,10 +89,21 @@ def all_gather(self, tensor: Tensor, group: Optional[Any] = None, sync_grads: bo
         """Perform a all_gather on all processes."""
         return all_gather_ddp_if_available(tensor, group=group, sync_grads=sync_grads)
 
-    def reduce_boolean_decision(self, decision: bool) -> bool:
+    def reduce_boolean_decision(self, decision: bool, all: bool = True) -> bool:
+        """Reduces a boolean decision over distributed processes. By default is analagous to ``all`` from the
+        standard library, returning ``True`` only if all input decisions evaluate to ``True``. If ``all`` is set to
+        ``False``, it behaves like ``any`` instead.
+
+        Args:
+            decision (bool): A single input decision.
+            all (bool, optional): Whether to logically emulate ``all`` or ``any``. Defaults to True.
```

Comment:
```suggestion
            decision: A single input decision.
            all: Whether to logically emulate ``all`` or ``any``. Defaults to True.
```
Nit: In our code base, we don't explicitly list the types here. We rely on the typing in the signature, and the docs generator to list the types for us.